draft-ietf-wimse-client-assertion-in-workload-environments.md

```diff
@@ -0,0 +1,546 @@
+"SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and
+"OPTIONAL" in this document are to be interpreted as described in
+BCP 14 {{RFC2119}} {{RFC8174}} when, and only when, they appear in all
+capitals, as shown here.
+
+For the scope of this specification, the term authorization server is only used for the authorization server of the external authorization domain as highlighted in {{fig-overview}}.
+
+Even though, technically, the platform credential is also issued by an authorization server within the workload platform, this specification only refers to it as "platform issuer" or just "platform".
+
+# OAuth Assertion in Workload Environments
+
+## Overview
+
+{{fig-overview}} illustrates a generic pattern that applies across all of the patterns described in {{patterns}}:
+
+~~~ aasvg
+ +----------------------------------------------------------+
+ |                           External Authorization Domain  |
+ |                                                          |
+ | +-------------------------+     +--------------------+   |
+ | |                         |     |                    |   |
+ | |   Authorization Server  |     | Protected Resource |   |
+ | |                         |     |                    |   |
+ | +-------^---------+-------+     +----------^---------+   |
+ +---------+---------+------------------------+-------------+
+           |         |                        |
+           |   3) access token           4) access
+           |         |                        |
+2) present assertion |                        |
+           |         |     +------------------+
+           |         |     |
+ +---------+---------+-----+--------------------------------+
+ |         |         |     |             Workload Platform  |
+ |         |         |     |                                |
+ |  +------+---------v-----+---+           +-------------+  |
+ |  |                          |           | Credential  |  |
+ |  |        Workload          +-----------> issued by   |  |
+ |  |                          |  1) get   | Platform    |  |
+ |  +--------------------------+           +-------------+  |
+ +----------------------------------------------------------+
+~~~
+{: #fig-overview title="OAuth2 Assertion Flow in generic Workload Environment"}
+
+The figure outlines the following steps which are applicable in any pattern.
+
+* 1) retrieve credential issued by platform. The way this is achieved and whether this is workload or platform initiated differs based on the platform.
+
+* 2) present credential as an assertion towards an authorization server in an external authorization domain. This step uses the assertion_grant flow defined in {{RFC7521}} and, in case of JWT format, {{RFC7523}}.
+
+* 3) On success, an access token is returned to the workload to access the protected resource.
+
+* 4) The access token is used to access the protected resource in the external authorization domain.
```

Comment:
This terminology is very confusing to first-timers. Please say, at least once, that when we discuss "accessing a resource" we also include "sending a message" or "making a REST call".